PopeSpaceous/CodeTalker
# Migrate speech to text functionality to a docker container

Closes #13 

## What was Done:
- Took @steaward's code and modified some of the directory paths it specified to make it more portable. 
- Moved all microphone code from extension.js and created its own dedicated node module in micContainer/containerFiles. This directory now gets mounted to a docker container when the extension is run in VS Code.

## Important notes:
Current implementation only supports unix style machines since it mounts the /dev/snd directory to the docker container to pass in microphone functionality. This works because Unix systems mount hardware to their directory trees. **This was tested and run on Ubuntu 18.04**

## To test:
- **Start by installing Docker** if you don't have it on your system already: [Installation instructions for a variety of systems can be found in this document](https://docs.docker.com/install/#supported-platforms_)

- **Build the Docker image:** Navigate to CodeTalker/micContainer, and execute the command:
> docker build -t mic-container .

Don't forget the "." at the end of this command since it indicates you want to use the Dockerfile in the current directory. The first build will take a few minutes, but any later builds will be faster since Docker caches parts of the build process.

- in VS code, hit F5, go to the command pallette in the new window that came up, and enter "Code Talker" (that's right; it's no longer "Hello World"). Speech to text output should appear in the debug console output of the old window.
